src/server/scripts/Northrend/AzjolNerub/AzjolNerub/boss_krikthir_the_gatewatcher.cpp

```diff
@@ -116,7 +116,31 @@ class boss_krik_thir : public CreatureScript
 
         struct boss_krik_thirAI : public BossAI
         {
-            boss_krik_thirAI(Creature* creature) : BossAI(creature, DATA_KRIKTHIR_THE_GATEWATCHER) { }
+            public:
+                boss_krik_thirAI(Creature* creature) : BossAI(creature, DATA_KRIKTHIR_THE_GATEWATCHER), _hadGreet(false) { }
+
+                void DoAction(int32 action) override
+                {
+                    switch (action)
+                    {
+                        case -ACTION_GATEWATCHER_GREET:
```

Comment:
I use negative values for actions defined in the instance script's header, and positive values for actions originating in the boss script.